spec/models/miq_provision_workflow_spec.rb

```diff
@@ -53,20 +61,77 @@
         it "should encrypt fields" do
           password_input = "secret"
           request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
-            "1.1", admin, "name=template", "vm_name=spec_test|root_password=#{password_input}",
-            "owner_email=admin|owner_first_name=test|owner_last_name=test", nil, nil, nil, nil)
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test', 'root_password' => "#{password_input}"},
+            {'owner_email' => 'admin'}, {'owner_first_name' => 'test'},
+            {'owner_last_name' => 'test'}, nil, nil, nil, nil)
 
           MiqPassword.encrypted?(request.options[:root_password]).should be_true
           MiqPassword.decrypt(request.options[:root_password]).should == password_input
         end
 
+        it "should set values when extra '|' are passed in" do
+          request = ManageIQ::Providers::Vmware::InfraManager::ProvisionWorkflow.from_ws(
+            "1.1", admin, {'name' => 'template'}, {'vm_name' => 'spec_test'},
+            nil, nil, {'abc' => 'tr|ue', 'blah' => 'nah'}, nil, nil)
+
+          expect(request.options[:ws_values]).to include(:abc => "tr|ue")
+        end
+
+        it "should set values when extra '|' are passed in for multiple values" do
```

Comment:
@syncrou Is one test enough to validate parsing of the `|` symbol?  Is there any expectation that the number of parameters passed will cause a different result?  This seems like duplicate tests and I would lean towards just keeping this one.